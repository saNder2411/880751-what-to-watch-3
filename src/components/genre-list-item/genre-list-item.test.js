import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import GenreListItem from './genre-list-item';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);
const store = mockStore({});
const label = `All Genre`;
const isActive = true;
const onGenreListItemClick = () => {};


it(`Should GenreListItem render correctly`, () => {
  const markup = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <Switch>
              <Route
                path='/'
                render={() => {
                  return <GenreListItem
                    label={label}
                    isActive={isActive}
                    onGenreListItemClick={onGenreListItemClick}/>;
                }}
              />
            </Switch>
          </BrowserRouter>
        </Provider>
    )
    .toJSON();

  expect(markup).toMatchSnapshot();
});
