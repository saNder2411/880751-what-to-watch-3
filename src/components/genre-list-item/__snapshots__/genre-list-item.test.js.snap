// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenresListItem render correctly 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    All Genre
  </a>
</li>
`;
